File: backend/src/main/java/magazyn/dto/KonfiguracijaDTO.java
package magazyn.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class KonfiguracijaDTO {

    private Integer id;

    @NotBlank(message = "Nazwa parametru jest wymagana")
    private String nazwaParametru;

    @NotBlank(message = "Wartość parametru jest wymagana")
    private String wartoscParametru;

    private String typParametru; // STRING, INTEGER, BOOLEAN, DECIMAL

    private String opis;

    private Boolean aktywna = true;
}

File: frontend/app/src/main/java/com/example/magazyn/api/dtos/KonfiguracijaDTO.kt
package com.example.magazyn.api.dtos

data class KonfiguracijaDTO(
    val id: Int? = null,
    val nazwaParametru: String,
    val wartoscParametru: String,
    val typParametru: String = "STRING",
    val opis: String = "",
    val aktywna: Boolean = true
)
